clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -1120,6 +1125,78 @@ void LoweringPreparePass::lowerIterEndOp(IterEndOp op) {
   op.erase();
 }
 
+void LoweringPreparePass::lowerStdInitializerListOp(StdInitializerListOp op) {
+  auto loc = op.getLoc();
+  cir::CIRDataLayout dataLayout(theModule);
+  auto ptrDiffTy = cir::IntType::get(&getContext(), 64, /*isSigned=*/false);
+
+  auto args = op.getArgs();
+
+  auto stdInitializerListType =
+      mlir::cast<cir::StructType>(op.getInitList().getType().getPointee());
+  clang::RecordDecl::field_range stdInitializerListFields =
+      stdInitializerListType.getAst().getRawDecl()->fields();
+  assert(llvm::range_size(stdInitializerListFields) == 2U);
+
+  mlir::Type elementType =
+      mlir::cast<cir::PointerType>(stdInitializerListType.getMembers()[0])
+          .getPointee();
+  auto tempArrayType =
+      cir::ArrayType::get(&getContext(), elementType, args.size());
+  auto alignment = mlir::IntegerAttr::get(
+      mlir::IntegerType::get(&getContext(), 64),
+      dataLayout.getPrefTypeAlign(tempArrayType).value());
+
+  CIRBaseBuilderTy builder(getContext());
+  builder.setInsertionPointAfter(op);
+
+  mlir::Value arrayPtr = builder.createAlloca(
+      loc, cir::PointerType::get(tempArrayType), tempArrayType, "", alignment);
+  mlir::Value arrayStartPtr =
+      builder.createCast(cir::CastKind::array_to_ptrdecay, arrayPtr,
+                         cir::PointerType::get(elementType));
+  for (unsigned i = 0; i < args.size(); i++) {
+    if (i == 0) {
+      builder.createStore(loc, args[i], arrayStartPtr);
+    } else {
+      mlir::Value offset =
+          builder.create<cir::ConstantOp>(loc, cir::IntAttr::get(ptrDiffTy, i));
+      mlir::Value dest = builder.create<cir::PtrStrideOp>(
+          loc, arrayStartPtr.getType(), arrayStartPtr, offset);
+      builder.createStore(loc, args[i], dest);
+    }
+  }
+
+  // FIXME(cir): better handling according to different field type. [ptr ptr],
+  // [ptr size], [size ptr].
+
+  clang::RecordDecl::field_iterator stdInitializerListField =
+      stdInitializerListFields.begin();
+  mlir::Value startMemberPtr = builder.createGetMemberOp(
+      loc, op.getInitList(), stdInitializerListField->getName().data(), 0);
+  builder.createStore(loc, arrayStartPtr, startMemberPtr);
+
+  ++stdInitializerListField;
```

Comment:
It's not a good idea to introduce mutable iterator for a 2-tuple. Please name the two elements at the beginning.